app/screens/permalink/permalink.js

```diff
@@ -359,7 +397,7 @@ export default class Permalink extends PureComponent {
                     onHashtagPress={onHashtagPress}
                     onPermalinkPress={onPermalinkPress}
                     onPostPress={this.goToThread}
-                    postIds={postIds}
+                    postIds={postIdsState}
```

Comment:
When post is deleted, we're passing `postIdsState` which holds previous postIds (including the deleted `focusedPostId`).  This causes the `makePreparePostIdsForPostList` to have a null `post` and so we added filter to remove that.